echopop/live/live_spatial_methods.py

```diff
@@ -0,0 +1,665 @@
+        dataset.update(
+            {k: apply_inpfc_definitions(d, spatial_dict["strata"]) for k, d in dataset.items()}
+        )
+    elif isinstance(dataset, pd.Series) and link_method == "INPFC":
+        return apply_inpfc_definitions(dataset, spatial_dict["strata"])
+
+
+# def apply_inpfc_definitions(acoustic_data: dict, biology_data: dict, spatial_config: dict):
+
+#     # Extract the INPFC definitions
+#     inpfc_definitions = spatial_config["inpfc"]
+
+#     # Create latitude bins
+#     latitude_bins = np.concatenate([[-90.0], inpfc_definitions["latitude_max"], [90.0]])
+#     # ---- Append 1 more stratum layer
+#     bin_names = np.concatenate([inpfc_definitions["stratum_names"],
+#                                 [np.max(inpfc_definitions["stratum_names"]) + 1]])
+
+#     # Create spatial key
+#     spatial_config["spatial_key"] = pd.DataFrame({
+#         "latitude_limit": inpfc_definitions["latitude_max"],
+#     })
+#     # ---- Cut
+#     spatial_config["spatial_key"]["stratum"] = (
+#         pd.cut(inpfc_definitions["latitude_max"],
+#                latitude_bins,
+#                right = True,
+#                labels = bin_names)
+#     )
+
+#     # Get the `prc_nasc_df` values, if they exist, and apply stratification information
+#     if not acoustic_data["prc_nasc_df"].empty:
+#         # ---- Bin the latitude data
+#         acoustic_data["prc_nasc_df"]["stratum"] = pd.cut(
+#             acoustic_data["prc_nasc_df"]["latitude"],
+#             latitude_bins,
+#             right = True,
+#             labels = bin_names,
+#         )
+
+#     # Get the `trawl_info_df` values, if they exist, and apply stratification information
+#     if not biology_data["trawl_info_df"].empty:
+#         # ---- Bin the latitude data
+#         biology_data["trawl_info_df"]["stratum"] = pd.cut(
+#             biology_data["trawl_info_df"]["latitude"],
+#             latitude_bins,
+#             right = True,
+#             labels = bin_names,
+#         )
+
+
+def define_boundary_box(boundary_dict: dict, projection: str):
+
+    # Get x-coordinates
+    if "longitude" in boundary_dict.keys():
+        x = np.array(boundary_dict["longitude"])
+    else:
+        x = np.array(boundary_dict["northings"])
+
+    # Get y-coordinates
+    if "latitude" in boundary_dict.keys():
+        y = np.array(boundary_dict["latitude"])
+    else:
+        y = np.array(boundary_dict["eastings"])
+
+    # Create a boundary DataFrame
+    bound_df = pd.DataFrame(
+        {
+            "x": np.array([x.min(), x.max(), x.max(), x.min(), x.min()]),
+            "y": np.array([y.min(), y.max(), y.max(), y.min(), y.min()]),
+        }
+    )
+
+    # Convert to a GeoDataFrame and return the GeoDataFrame
+    return gpd.GeoDataFrame(
+        data=bound_df,
+        geometry=gpd.points_from_xy(bound_df["x"], bound_df["y"]),
+        crs=projection,
+    )
+
+
+def apply_griddify_definitions(dataset: pd.DataFrame, spatial_config: dict):
```

Comment:
I think the first half of `apply_griddify_definitions` (up to # Convert to GeoDataFrame where you actually handles the acoustic data) could be separate out as a function that you run once in the init and store the output grids needed to "assign" the NASC data into.